imports/wasi_snapshot_preview1/fs_test.go

```diff
@@ -3937,6 +3937,30 @@ func Test_pathOpen(t *testing.T) {
 			expectedLog: `
 ==> wasi_snapshot_preview1.path_open(fd=3,dirflags=,path=file,oflags=,fs_rights_base=FD_READ|FD_WRITE,fs_rights_inheriting=,fdflags=)
 <== (opened_fd=4,errno=ESUCCESS)
+`,
+		},
+		{
+			name:          "sysfs.DirFS file O_DIRECTORY RIGHTS_FD_WRITE",
```

Comment:
@evacchi This test ensures the erroneous `isdir` is not returned for regular file and the `O_DIRECTORY` flag is set.